### PR TITLE
Update spec and remove all deprecated info

### PR DESCRIPTION
**Pull request by majido**
This is a pretty large patch. It is a major rewrites of most of the spec including the following 
changes:
- Remove any deprecated info
- Add structure to support the new specification with appropriate TODO notes for missing information
- Update Animator Definition and Animation concepts to match the new design
- Update Animation Definition registration and Animator Instance creation/running algorithms.
- Update all example using the EXPLAINER syntax

I will follow up this with other patches that will replace TODO with more accurate descriptions.

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -194,57 +185,52 @@ interface AnimationWorkletGlobalScope : WorkletGlobalScope {
...
+ - A <dfn>class constructor</dfn> which is the class <a>constructor</a>.
 
- - An <dfn>animator slot</dfn> <<ident>>#.
+ - An <dfn>animate function</dfn> which is the animate <a>function</a> callback.
 
- - An <dfn>animator input property list</dfn>.
 
- - An <dfn>animator output property list</dfn>.
 
+An <dfn>animator instance</dfn> a <a>struct</a> which describes a fully realized custom animation
```
Comment (stephenmcgruer) [inline index.bs:199]: 'is a \<a\>struct\</a\>' ?

File: index.bs [hunk 2]
```diff
@@ -267,12 +253,6 @@ When the <dfn method for=AnimationWorkletGlobalScope>registerAnimator(|name|,
     7. If the result of <a>IsCallable</a>(|animate|) is false, <a>throw</a> a <a>TypeError</a> and
         abort all these steps.
 
-    8. Let |elements| be the result of <a>parsing an animator slot list</a> for |animatorCtor|.
-        If an exception is thrown, rethrow the exception and abort all these steps.
-
-    9. Let |timelineOptions| be the result of <a>parsing an animator timeline options list</a> for
-        |animatorCtor|. If an exception is thrown, rethrow the exception and abort all these steps.
-
     10. Let |definition| be a new <a>animator definition</a> with:
```
Comment (stephenmcgruer) [inline index.bs:256]: Should this be re-numbered to '8' ?

**Review by stephenmcgruer — APPROVED** [review]

File: index.bs (hunk 2 above)
Comment (majido) [reply index.bs:256]: I will fix in a follow up patch.

File: index.bs (hunk 1 above)
Comment (majido) [reply index.bs:199]: I will fix this in a follow up patch.

**Pull request merged**